From 996d92d2710f9dc740351f4d9cbe14af64569689 Mon Sep 17 00:00:00 2001
From: Mikhail Gusarov <dottedmag@dottedmag.net>
Date: Fri, 07 May 2010 20:22:20 +0000
Subject: Get rid of Xalloc/Xrealloc/Xfree from X server or Xlib

[ RHEL partial revert for ABI compliance ]

alloc/realloc/free calls are encapsulated in libXdmcp, so
there is no need to wrap allocation functions even under Windows

Signed-off-by: Mikhail Gusarov <dottedmag@dottedmag.net>
Signed-off-by: Alan Coopersmith <alan.coopersmith@oracle.com>
---
diff --git a/Alloc.c b/Alloc.c
deleted file mode 100644
index f180eaf..0000000
--- a/Alloc.c
+++ b/dev/null
@@ -1,74 +0,0 @@
-/*
-Copyright 1989, 1998  The Open Group
-
-Permission to use, copy, modify, distribute, and sell this software and its
-documentation for any purpose is hereby granted without fee, provided that
-the above copyright notice appear in all copies and that both that
-copyright notice and this permission notice appear in supporting
-documentation.
-
-The above copyright notice and this permission notice shall be included in
-all copies or substantial portions of the Software.
-
-THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
-OPEN GROUP BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN
-AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
-CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
-
-Except as contained in this notice, the name of The Open Group shall not be
-used in advertising or otherwise to promote the sale, use or other dealings
-in this Software without prior written authorization from The Open Group.
- * *
- * Author:  Keith Packard, MIT X Consortium
- */
-
-/* stubs for use when Xalloc, Xrealloc and Xfree are not defined */
-
-#ifdef HAVE_CONFIG_H
-#include <config.h>
-#endif
-#include <X11/Xos.h>
-#include <X11/X.h>
-#include <X11/Xmd.h>
-#include <X11/Xdmcp.h>
-#include <stdlib.h>
-
-/* this probably works for Mach-O too, but probably not for PE */
-#if defined(__ELF__) && defined(__GNUC__) && (__GNUC__ >= 3)
-#define weak __attribute__((weak))
-#else
-#define weak
-#endif
-
-#ifdef __SUNPRO_C
-#pragma weak Xalloc
-#pragma weak Xrealloc
-#pragma weak Xfree
-#endif
-
-weak void *
-Xalloc (unsigned long amount)
-{
-    if (amount == 0)
-	amount = 1;
-    return malloc (amount);
-}
-
-weak void *
-Xrealloc (void *old, unsigned long amount)
-{
-    if (amount == 0)
-	amount = 1;
-    if (!old)
-	return malloc (amount);
-    return realloc ((char *) old, amount);
-}
-
-weak void
-Xfree (void *old)
-{
-    if (old)
-	free ((char *) old);
-}
diff --git a/Makefile.am b/Makefile.am
index 4a55544..244a0c2 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -8,7 +8,6 @@ AM_CFLAGS = \
 libXdmcp_la_LDFLAGS = -version-number 6:0:0 -no-undefined
 libXdmcp_la_LIBADD = $(XDMCP_LIBS)
 libXdmcp_la_SOURCES = \
-	Alloc.c \
 	Array.c \
 	Fill.c \
 	Flush.c \
--
cgit v0.9.0.2-2-gbebe
